### PR TITLE
Armor Rendering Registry: Render Layers

### PR DESCRIPTION
**Pull request by Leo40Git**
An additional feature for the Armor Rendering Registry, based off of [Yttr's code](https://git.sleeping.town/unascribed/Yttr/src/commit/f2d7e35909bcb21beeb4cdbfe51aa072f24331d6/src/main/java/com/unascribed/yttr/mixin/ultrapure/client/MixinArmorFeatureRenderer.java#L110).

### CODE REVIEW

**Comment by Leo40Git** [comment]
Converting to draft for now, pending utility methods for registering non-dynamic custom models/render layers.

**Comment by EnnuiL** [comment]
Hm, I assume that the requirements for getting this out of draft have been met?

**Comment by Leo40Git** [comment]
oops

**Comment by EnnuiL** [comment]
Hm, at this point, something tells me it would be better to rebase all Armor Rendering Registry work on 1.19.4

**Comment by Leo40Git** [comment]
oh right I forgot about this and Mojang said "y'know what let's break armor"
Will try to rebase this onto 1.19.4 tomorrow-ish, I need to get my dev environment set up.

**Comment by Leo40Git** [comment]
Rebased to 1.19.4! Apparently I slightly screwed up the commit history though, oops

File: library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java [hunk 1]
```diff
@@ -82,25 +79,87 @@ public abstract class ArmorFeatureRendererMixin {
...
+	@ModifyArg(
+			method = "renderArmor(Lnet/minecraft/item/ArmorMaterial;Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/client/render/VertexConsumerProvider;ILnet/minecraft/item/trim/ArmorTrimPermutation;ZLnet/minecraft/client/render/entity/model/BipedEntityModel;ZFFF)V",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/client/texture/SpriteAtlasTexture;getSprite(Lnet/minecraft/util/Identifier;)Lnet/minecraft/client/texture/Sprite;"
+			),
+			index = 0
+	)
+	private Identifier quilt$modifyArmorTrimTexture(Identifier texture) {
+		// TODO
```
Comment (EnnuiL) [inline library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java:138]: hm, what is this TODO?

**Review by EnnuiL — APPROVED** [review]
Looks good!

File: library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java (hunk 1 above)
Comment (Leo40Git) [reply library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java:138]: Armor trim texture modification, possibly.

Comment (EnnuiL) [reply library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java:138]: ah

**Comment by Leo40Git** [comment]
**TODO:** Armor Material Registry API for armor trim shenanigans.

**Review by TheGlitch76 — DISMISSED** [review]
If this is based off of yttr code this code either needs attribution as required by [Yttr's license](https://git.sleeping.town/unascribed/Yttr/src/commit/f2d7e35909bcb21beeb4cdbfe51aa072f24331d6/LICENSE) or @unascribed's blessing

**Comment by EnnuiL** [comment]
> If this is based off of yttr code this code either needs attribution as required by [Yttr's license](https://git.sleeping.town/unascribed/Yttr/src/commit/f2d7e35909bcb21beeb4cdbfe51aa072f24331d6/LICENSE) or @unascribed's blessing

Una has given her blessing [on Greencord](https://discord.com/channels/833872081585700874/905865836231811142/1068940150207414443)

**Review by TheGlitch76 — APPROVED** [review]

**Comment by EnnuiL** [comment]
Conflict with the fixed mixin needs to be fixed!

**Comment by TheGlitch76** [comment]
@Leo40Git can you resolve the conflicts on this PR?

**Comment by Leo40Git** [comment]
oops, lemme do that rq

**Comment by Leo40Git** [comment]
@EnnuiL @TheGlitch76 Conflicts resolved!

I'm considering the TODO for modifying armor trim textures (AKA custom armor trims) unresolvable until we have an ArmorMaterial registry, which I may claim later...

**Review by EnnuiL — APPROVED** [review]
I believe all that is needed is yeah, testing this API in order to see if it'll work :p

**Comment by EnnuiL** [comment]
This PR has entered a Final Comment Period of 5 days!

**Review by LambdAurora — APPROVED** [review]

**Pull request merged**